src/authorization/UrlBasedAclManager.ts

```diff
@@ -23,4 +14,13 @@ export class UrlBasedAclManager implements AclManager {
   public async isAcl(id: ResourceIdentifier): Promise<boolean> {
     return /\.acl\/?/u.test(id.path);
   }
+
+  public async getResource(aclId: ResourceIdentifier): Promise<ResourceIdentifier> {
+    if (!await this.isAcl(aclId)) {
```

Comment:
This condition does not match the function's description in the interface (and the suggested variable naming), where `aclId` is always an ACL.

Reply:
Any preference which way to go with this? Either works (changing the function description or what it actually does). I think I did this to be consistent with the `getAcl` function.

Reply:
Let's keep the definition, but have the signature as

```
  public async getAclConstrainedResource(id: ResourceIdentifier): Promise<ResourceIdentifier> {
```

The idea being that we get the identifier of the resource on which the ACL constraints should be placed.

and I think `getAcl` should become `getAclDocument` (and also `isAclDocument`).

The two functions appear to be symmetrical, but they are not; it is a coincidence. The first says what the ACL document is for a given resource; the second on what resource constraints should be placed.